src/actions/public/L2/getWithdrawalMessages.ts

```diff
@@ -20,9 +21,9 @@ export type GetWithdrawalMessagesReturnType = {
  */
 export async function getWithdrawalMessages<TChain extends Chain | undefined>(
   client: PublicClient<Transport, TChain>,
-  { hash }: GetWithdrawalMessagesParameters,
+  { hash, txReceipt }: GetWithdrawalMessagesParameters,
 ): Promise<GetWithdrawalMessagesReturnType> {
-  const receipt = await client.getTransactionReceipt({ hash })
+  const receipt = txReceipt ?? await client.getTransactionReceipt({ hash })
```

Comment:
It's a nit but I was gonna make it only take a txReceipt and never a hash. I have always hated this behavior from the sdk. Don't feel strongly though and sure others like being able to pass in both